### ISSUE TITLE
Cannot start client with checkpoint sync

### ISSUE THREAD

**Issue by alexpeterson91** [dappnode/DAppNodePackage-Lodestar-Prater#58]
@nflaig 
I installed from the published version, I manually changed the global ENVs in the EL package im using and the W3S prater package so its all working there save for integration in stakersUI, but this is confusing me a lot why it wont accept the default checkpoint sync endpoint we use for Prater, hardcoded into the package. and could be a high priority blocking issue.

I cannot start the Lodestar Prater Client with checkpoint sync, it begins a restart loop and throws these errors, it doesn't show up as a card on the dashboard in this state or in the chains status bar.
[logs-dappnode-package-DAppNodePackage-beacon-chain.lodestar-prater.dnp.dappnode.eth.txt](https://github.com/dappnode/DAppNodePackage-Lodestar-Prater/files/11044648/logs-dappnode-package-DAppNodePackage-beacon-chain.lodestar-prater.dnp.dappnode.eth.txt)

I then removed the checkpoint sync restarted it and no more errors just very very slow syncing. This time the card shows properly on the dashboard and in the chains status dropdown. logs below.
[logs-dappnode-package-DAppNodePackage-beacon-chain.lodestar-prater.dnp.dappnode.eth (no-checkpoint).txt](https://github.com/dappnode/DAppNodePackage-Lodestar-Prater/files/11044658/logs-dappnode-package-DAppNodePackage-beacon-chain.lodestar-prater.dnp.dappnode.eth.no-checkpoint.txt)

**Comment by nflaig** [comment]
I just tested this with different checkpoint sync endpoints and it seems to fail in some cases, related issue https://github.com/ChainSafe/lodestar/issues/4524

**Comment by nflaig** [comment]
> I cannot start the Lodestar Prater Client with checkpoint sync, it begins a restart loop and throws these errors, it doesn't show up as a card on the dashboard in this state or in the chains status bar.

It looks like the checkpoint sync fails if the slot on the epoch boundary does not contain a block. If that happens, a user has to wait for the next epoch to hopefully resolve the problem, see related issue https://github.com/ChainSafe/lodestar/issues/4524.

It will likely not happen on mainnet as it directly related to low network participation on goerli.

The checkpoint sync endpoint is also only fetched if there is not already a initialized DB, so it really can only happen on first installation and will resolve itself after a while automatically.


It definitely works with `https://checkpoint-sync-prater.dappnode.io` as checkpoint sync url.
[logs-dappnode-package-DAppNodePackage-beacon-chain.lodestar-prater.dnp.dappnode.eth.txt](https://github.com/dappnode/DAppNodePackage-Lodestar-Prater/files/11049826/logs-dappnode-package-DAppNodePackage-beacon-chain.lodestar-prater.dnp.dappnode.eth.4.txt)

> I then removed the checkpoint sync restarted it and no more errors just very very slow syncing. This time the card shows properly on the dashboard and in the chains status dropdown. logs below.

yes, syncing from genesis takes quite a while and is not ideal due to [long range attacks](https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/attack-and-defense/#long-range-attacks).

**Comment by alexpeterson91** [comment]
well since lodestar is new for everyone on dappnode and we use checkpoint sync by default to get everyone going in a couple minutes, is this what everyone will expect on prater for lodestar? because i had it running for 2 hours in that restarting mode with the checkpoint sync set to the default endpoint as above, and it never changed the error message or anything, just a restart loop over and over, so i was by no means impatient and had waited many, many epochs, i know goerli is slow right now I've got like 350 prater validators across my nodes and I've been staring at the times the chain slows due to offline nodes and finalization being delayed
For every other Prater CL package, i clear the volumes regularly basically anytime i change nodes (which is often, since i run 6 Dappnodes at the moment and they're constantly in use for  testing new versions of packages, and new packages, I don't like to wait for CL client syncing to catch up after being paused even for a day while checking another client, so for teku prater, nimbus prater, lighthouse prater, prysm prater, i just wipe the volume let it re-sync from checkpoint and its always synced and ready to validate in 5 mins on average 3 mins, even now.  Never had a delay over 5 mins on any other client than lodestar prater so far with regards to checkpoint sync and never has it caused a restart loop in other clients. 
Im gonna spin it up on a brand new node i just installed and see what happens there, since i cant explain why i couldn't get it to work already on my oldest node, it just happened to have all 4 prater EL clients synced and seemed like the right place to test lodestar prater yesterday.

Obviously it doesnt seem that others testing are having this issue but i am, so I'm gonna try and see what could be wrong on my end, since it was installed signed and officially from the dappstore via ENS not via unsigned IPFS hash or a sdk direct build, as is necessary before it's been published.  So that is also a possibility why i have a different result, since i started testing lodestar prater just after it was published, but obviously you've been testing hash or sdk builds for the last several weeks since lodestar prater hasnt been published for much more than 24 hours now. but lets hope i just have a bug in that node lol. and my next attempt has no issues.

Separately i know syncing from genesis takes a very long time and isn't ideal, I've been around since before checkpoint sync was introduced or working on all clients etc.  I am curious though, last fall i proposed that we change some wording and make checkpoint sync default (obviously things have since changed and we do now encourage its use and warn when its toggled to disabled. and we have it set as default now), but at the time i got a lot of pushback from some users and the team explained the users' reluctance was that they wanted no 3rd parties involved in their syncing, and preferred to do it from genesis from a hyper-security perspective, so the Long Range attacks are not something ive even heard of until lodestar started without a checkpoint and mentioned it as you did above too. i guess i need to read up on that since i was given the impression that syncing from checkpoint is less secure, but far more convenient. Though it seems i was misled about the security of syncing from genesis. thanks for the link im gonna read up on it.

**Comment by nflaig** [comment]
Sorry that you had to deal with those issues, unfortunately I do not think this is related to any specifics in your setup. The issue is really a combination of how the checkpoint sync providers works and how Lodestar processes that data. In the issue on the Lodestar repo you can find more information about this, it is still not concluded if this is a problem in Lodestar or rather something checkpoint sync providers need to fix in the long-term.

Ideally, Lodestar should just work and be ready in under 5mins (like other CLs). Wiping the volume and re-sync from checkpoint needs to consistently work even on goerli.  Will definitely check this with the other engineers if we can achieve more stability in the checkpoint sync flow, I am also not sure if it is good that we exit the process, might be better to just show an error and retry internally with  exponential backoffs—i.e., a delay of 10 seconds, 20 seconds, 40 seconds, and so on for up to 6 minutes (or next expected epoch). Progress on this can be checked here https://github.com/ChainSafe/lodestar/issues/4524.

Regarding the sync from genesis vs checkpoint, I am far from an expert on this and there are several things to consider. We have a section about [Checkpoint Sync](https://chainsafe.github.io/lodestar/usage/beacon-management/#checkpoint-sync) in the Lodestar docs, it is possible to provide an additional `wssCheckpoint` to cross-check the data fetched from the checkpoint sync provider, I would say this is the best option to achieve the highest security. If you want to really dig into this, it might be worth giving Vitalik's post [Proof of Stake: How I Learned to Love Weak Subjectivity](https://blog.ethereum.org/2014/11/25/proof-stake-learned-love-weak-subjectivity) a read.

**Comment by philknows** [comment]
This unfortunately does happen often on Goerli because of the nothing-at-stake problem and many missed blocks cause this to be an issue quite frequently. I'm not quite familiar enough with the other clients to understand what they may be doing to mitigate this if you're wiping your volumes regularly and need to checkpoint sync quite regularly, but if you're noticing that it's always Lodestar that will not be able to grab and process a state when the other 4 clients can from the same checkpoint at the same time, I'd definitely note this in our issue.

**Comment by g11tech** [comment]
This is not a problem with lodestar, checkpointz software does not cache the states where finalized/checkpoint 's block root is not perfecly aligned with the epoch boundary.

if however the current  finalized checkpoint is such, and one doesn't provide any explicit checkpoint published by checkpointz then this problem will occur as lodestar will try to load the finalized state which checkpointz obviously does NOT has.

**Comment by alexpeterson91** [comment]
> I'm not quite familiar enough with the other clients to understand what they may be doing to mitigate this if you're wiping your volumes regularly and need to checkpoint sync quite regularly, but if you're noticing that it's always Lodestar that will not be able to grab and process a state when the other 4 clients can from the same checkpoint at the same time, I'd definitely note this in our issue.

Sorry for delayed reply just found this wasnt sent and was just still open in my browser but this definitely was the case for me no issues with the other 4 CLs, (but timing with Shapella on goerli is certainly a possibility why I've been encountering it, just odd that i have no issue switching back and forth with the other prater/goerli CLs wiping their volumes using the same checkpoint sync (our endpoint for Dappnode) with no issues.  But i generally have the worst luck ever when it comes to stuff like this, if it can go wrong, it will go wrong for me first, its great for testing and catching some bugs early but in this case i think its just me being unlucky, will test again on another machine shortly, but consider this issue closed unless i report another issue

**Comment by nflaig** [comment]
>  if it can go wrong, it will go wrong for me first, its great for testing and catching some bugs early

that sounds great, maybe should consider pushing Lodestar pre-releases to dappnode as well so you can give it a try 😄

We will publish a patch release just for this soon (@g11tech fixed the issue), see https://github.com/ChainSafe/lodestar/pull/5319.

Probably should also publish another Lodestar Prater package once 1.7.1 update is merged

**Comment by philknows** [comment]
Yep, we got 1.7.1 in RC status now (should be out very quick if there's no issues detected) which includes that fix specifically since many people are encountering it and blocking users from trying Lodestar due to the headaches of getting setup on Goerli. Hope this resolves it for you @alexpeterson91 .
